format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSourceConfig.java

```diff
@@ -60,28 +61,33 @@
   @Macro
   private Long maxSplitSize;
 
+  @Macro
   @Nullable
   @Description("Whether to allow an input that does not exist. When false, the source will fail the run if the input "
     + "does not exist. When true, the run will not fail and the source will not generate any output. "
     + "The default value is false.")
   private Boolean ignoreNonExistingFolders;
 
+  @Macro
   @Nullable
   @Description("Whether to recursively read directories within the input directory. The default is false.")
   private Boolean recursive;
 
   @Name(PATH_FIELD)
+  @Macro
   @Nullable
   @Description("Output field to place the path of the file that the record was read from. "
     + "If not specified, the file path will not be included in output records. "
     + "If specified, the field must exist in the output schema as a string.")
   private String pathField;
 
+  @Macro
```

Comment:
have we tested a pipeline where all of these are macros? Have seen bugs introduced in the past where the property is assumed to not be a macro and a null pointer exception is generated at configure time.

Reply:
yes, I tested a File source -> File sink pipeline where all these fields are macros.